extensions/content-manager/controllers/collection-types.js

```diff
@@ -142,38 +143,73 @@ module.exports = {
   },
 
   async update(ctx) {
-    const { userAbility, user } = ctx.state;
-    const { id, model } = ctx.params;
-    const { body } = ctx.request;
-
-    const entityManager = getService('entity-manager');
-    const permissionChecker = getService('permission-checker').create({ userAbility, model });
-
-    if (permissionChecker.cannot.update()) {
-      return ctx.forbidden();
+    try {
+      const { userAbility, user } = ctx.state;
+      const { id, model } = ctx.params;
+      const { body } = ctx.request;
+
+      if(model === 'application::project.project' ){
+        const { title, team } = body;
+        const project = await strapi.services.project.findOne({ id });
+        const formatedEmail = strapi.services['google-manager'].formatEmail(title);
+        const group = await strapi.services['google-manager'].getGroup(`${formatedEmail}@devlaunchers.com`);
+
+        //lets leaders join google group and gives them owner acl for calendar
+        team.leaders.forEach(async (leader) => {
```

Comment:
I think we've done this multiple times. Maybe we can make it a function?

Reply:
Are you talking about the loop part? or the whole things?

Reply:
Both. Which one makes sense to you to put in a function?

Reply:
I do kind of repeat the whole thing a couple of times. It might make sense to put all of it in a function? What's your take?

Reply:
Makes sense to me.

Reply:
cool I'm thinking about making it a project service since we're giving google resources to projects